class/practice_2.2/main.cpp

```diff
@@ -0,0 +1,126 @@
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <vector>
+#include <string>
+
+using namespace std;
+
+struct Pixel {
+    int r, g, b;
+
+    bool operator==(const Pixel& other) const {
+        return r == other.r && g == other.g && b == other.b;
+    }
+};
+
+const int WIDTH = 5;
+const int HEIGHT = 5;
+
+bool readImage(const string& filename, vector<vector<Pixel>>& image, const string& outputFilename) {
```

Comment:
having out parameter is not a good idea and may cause a lot of errors or strange behavior in case of non-empty data in image object.

Also it is better to have image class to encapsulate the logic.
Another point is that if you have const predefined values then you can use std::array instead of vector and it will allow you to avoid unnecessary memory allocation on push backs.

using ImageRow = std::array<Pixel, COL_COUNT>;
std::optional<ImageRow> readRow(const std::string& line)
then you will be able to do imageRows[iRow] = std::move(/*some result object from optional*/) it will be effective from memory perspective and readability will be increased.